AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CounterFactory.cs

```diff
@@ -9,42 +9,57 @@
     internal class CounterFactory
     {
         /// <summary>
+        /// Gets metric alias to be the value given by the user.
+        /// </summary>
+        /// <param name="counterName">Name of the counter to retrieve.</param>
+        /// <param name="reportAs">Alias to report the counter.</param>
+        /// <returns>Alias that will be used for the counter.</returns>
+        private string GetCounterReportAsName(string counterName, string reportAs)
+        {
+            if (reportAs == null)
+                return counterName;
+            else
+                return reportAs;
+        }
+
+        /// <summary>
         /// Gets a counter.
         /// </summary>
         /// <param name="counterName">Name of the counter to retrieve.</param>
+        /// <param name="reportAs">Alias to report the counter under.</param>
         /// <returns>The counter identified by counterName</returns>
-        public ICounterValue GetCounter(string counterName)
+        public ICounterValue GetCounter(string counterName, string reportAs = null)
         {
             switch (counterName)
             {
                 case @"\ASP.NET Applications(??APP_W3SVC_PROC??)\Request Execution Time":
                     return new PerformanceCounterFromJsonGauge(
-                        counterName,
+                        GetCounterReportAsName(counterName, reportAs),
```

Comment:
you repeat this line over and over again. It would be better to call it once in the beginning and store into the local variable